test/tools/ossfuzz/README.md

```diff
@@ -0,0 +1,24 @@
+## Intro
+
+[oss-fuzz][1] is Google's fuzzing infrastructure that performs continuous fuzzing. What this means is that, each and every upstream commit is automatically fetched by the infrastructure and fuzzed.
+
+## What does this directory contain?
+
+To help oss-fuzz do this, we (as project maintainers) need to provide the following:
+
+- test harnesses: C/C++ tests that define the `LLVMFuzzerTestOneInput` API. This determines what is to be fuzz tested.
+- build infrastructure: (c)make targets per fuzzing binary. Fuzzing requires coverage and memory instrumentation of the code to be fuzzed.
+
+Test harnesses and cmake script is provided in this directory. Apart from this, fuzzing configuration may be needed to make customize the fuzzing experience per fuzzing binary. To do this, we maintain a file with the `.options` extension. This file defines the fuzzing configuration as follows:
+  - `close_fd_mask`: When equal to `3` this pipes console output (stdout/stderr) generated by the test harness to `/dev/null`
+  - `dict`: Name of the dictionary file to be used for fuzz testing. A [dictionary file][] contains a newline-seperated-block-quote-enclosed set of strings
+
+## What is libFuzzingEngine.a?
+
+`libFuzzingEngine.a` is a oss-fuzz-related dependency. It is present in the Dockerized environment in which Solidity's oss-fuzz code will be built.
+
+## Is this directory relevant for routine Solidity CI builds?
+
+No. This is the reason why the `add_subdirectory(ossfuzz)` cmake directive is nested under the `if (DEFINED ENV{OSSFUZZ})` predicate.
```

Comment:
I think we should still verify that this is compilable and does useful things (e.g. if somehow change Solidity's source such that the calls that are made here don't really do something, the fuzzer would never complain) as part of the CI. Is that possible somehow?

Reply:
Didn't get you here. If changes made to Solidity source in the future:

1. Break oss-fuzz build: oss-fuzz will automatically notify us about the failure
2. Reduce test coverage: oss-fuzz will *not* notify us directly. However, historical coverage data is available via the oss-fuzz web front-end for manual inspection.

Reply:
If it's not too much work, I would prefer this to happen in an automated way and before a pull request is merged.

Reply:
This may be another PR altogether because afaiu a new CI job may need to be set up to mirror oss-fuzz build process. I am tempted to postpone this effort until I have a decent understanding of the solidity CI system.